pkg/prometheus/promcfg.go

```diff
@@ -363,13 +363,13 @@ func buildExternalLabels(p *v1.Prometheus) yaml.MapSlice {
 // validateConfigInputs runs extra validation on the Prometheus fields which can't be done at the CRD schema validation level.
 func validateConfigInputs(p *v1.Prometheus) error {
 	if p.Spec.EnforcedBodySizeLimit != "" {
-		if err := operator.ValidateSizeField(p.Spec.EnforcedBodySizeLimit); err != nil {
+		if err := operator.ValidateSizeField(string(p.Spec.EnforcedBodySizeLimit)); err != nil {
```

Comment:
eventually we can get rid of the redundant validation. To handle the transition, we can remove it in a couple of releases (lets say in v0.58.0).

@PhilipGough thoughts? If we agree on this, we need a TODO here.

Reply:
Yeah, adding a TODO sounds like the safest thing to do in this case. It would be good to identify any similar cases and aim to remove them all in the single release.